### DISCUSSION TITLE
LDAP issue

### DISCUSSION THREAD

**Discussion by umarkhan99** [netbox-community/netbox#8046] [Q&A]
I am trying to integrate LDAP for our netbox and have followed  instructions on the website.
Installed django
REMOTE_AUTH_BACKEND = 'netbox.authentication.LDAPBackend' -> configuration file
created ldap_config.py file
But unable to login "Error: Please enter a correct username and password"

I tried to enable logging by copying the following to my configuration file

`LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'handlers': {
        'file': {
            'level': 'INFO',
            'class': 'logging.FileHandler',
            'filename': '/var/log/netbox.log',
        },
    },
    'loggers': {
        'django': {
            'handlers': ['file'],
            'level': 'INFO',
        },
    },
}`

and created a /var/log/netbox.log file which doesnt populate after trying to authenticate to the site. Any help would be greatly appreciated

**Comment by candlerb** [comment]
The logging issue might be to do with permissions.  Try logging to `/tmp` or another directory which you create and set the ownership to match netbox.

I found an old config which I haven't used for a while:

```
#LOGGING = {
#    'version': 1,
#    'disable_existing_loggers': False,
#    'handlers': {
#        'file': {
#            'level': 'INFO',
#            'class': 'logging.FileHandler',
#            'filename': '/opt/netbox/log/netbox.log',
#        },
#    },
#    'loggers': {
#        'django': {
#            'handlers': ['file'],
#            'level': 'INFO',
#        },
#        'netbox': {
#            'handlers': ['file'],
#            'level': 'INFO',
#        },
#    },
#}
```

**Comment by umarkhan99** [comment]
Still nothing....I changed permission for the log file to 777. FY I am using centos 7.9

**Reply by bitcollector1** [reply]
If you have not already, try restarting the NetBox service. I'm working my way through the LDAP troubleshooting now. Also make sure the directory and file is already created. Good Luck!

**Comment by dritier** [comment]
Just copy this to the end or your _ldap_config.py_
```
import logging, logging.handlers
logfile = "/opt/netbox/logs/django-ldap-debug.log"
my_logger = logging.getLogger('django_auth_ldap')
my_logger.setLevel(logging.DEBUG)
handler = logging.handlers.RotatingFileHandler(
logfile, maxBytes=1024 * 500, backupCount=5)
my_logger.addHandler(handler)
```

Ensure the directory _/opt/netbox/logs_ exists (it doesn't by default) and make the netbox user the owner with -> `chown netbox:netbox /opt/netbox/logs`

Currently I'm in the same boat, couldn't get it working but at least logging now works
